### ISSUE TITLE
Settings sync: manual sync flow feedback

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#102581]
Overall the flow is nice imho 👍 

**Dialog**

The dialog does not seem to reflect what the options are, esp. given "Sync Manually" is the default:

![image](https://user-images.githubusercontent.com/900690/87529286-0f726680-c68f-11ea-9b59-f00ead3f349b.png)

**Actions**

This one deserves explanation, maybe at the minimum it should have a "..." and then trigger a dialog explaining what happens.

![image](https://user-images.githubusercontent.com/900690/87529386-37fa6080-c68f-11ea-9152-ad40c3d2ab3c.png)

**Icons**

The "merge" icon imho is bad, initially I thought this would refresh the view to diff it again:

![image](https://user-images.githubusercontent.com/900690/87529527-6d9f4980-c68f-11ea-8e30-dd0290884b18.png)

For consistency I would keep the icons (Accept Remote, Accept Local) even when the file does not exist in either place. Otherwise it gets a bit hard to understand what is going on.

![image](https://user-images.githubusercontent.com/900690/87529656-97f10700-c68f-11ea-8c49-37550bd2a6e9.png)

**Flow**

Since either "Accept Local", "Accept Remote" or "Merge" can be destructive operations, I wonder if a dialog should appear each time you click it explaining you exactly what the operation does. This is similar to clicking the "Discard Changes" button in Git that will really delete any changes you made. 

![image](https://user-images.githubusercontent.com/900690/87530035-1c438a00-c690-11ea-96e5-3b6f6052b7e3.png)

**Bugs**

Clicking on a file does not preserve focus but moves it into the editor:

![image](https://user-images.githubusercontent.com/900690/87529842-dc7ca280-c68f-11ea-88ca-7ffe7a5c1d2b.png)

//cc @misolori

**Comment by sandy081** [comment]
> This one deserves explanation, maybe at the minimum it should have a "..." and then trigger a dialog explaining what happens.

Yeah this should not be there now and it seems out of context. With latest changes this has to go into sync activity view. And this shows a dialog.

**Comment by bpasero** [comment]
More feedback from the UX call:
* the "Sync Manually" should probably not be the default because it is an "Expert Level" UI (Merge would probably the best default)
* if we want to make the user go through each change and review it, then I think the UI needs to reflect that, e.g. I compare this with how I resolve git merge conflicts

**Comment by sandy081** [comment]
- Made Merge as default
- Changed UI to allow user to accept the change and also to revert the decision.
- Provided big buttons to turn on sync (which is enabled only after user accepts all changes) and also to cancel .

![Kapture 2020-07-16 at 19 58 33](https://user-images.githubusercontent.com/10746682/87768007-c996d900-c81b-11ea-89d5-0bd8441e3a9c.gif)

![Kapture 2020-07-16 at 19 22 40](https://user-images.githubusercontent.com/10746682/87768034-d1ef1400-c81b-11ea-9854-0e9eca3ef404.gif)

**Comment by bpasero** [comment]
@sandy081 feedback from today's insider. Overall this is getting a lot better for me to understand given latest UX changes. But I also suggested this flow so I may be a bit biased ;)

**Dialog**

![image](https://user-images.githubusercontent.com/900690/87801926-bc481180-c850-11ea-8a11-07df1d4ff557.png)

* `Sync Manually` => `Sync Manually...` so that it is clear that further input is needed
* I still feel the wording on that dialog only talks about `Sync` or `Merge` when actually `Sync Manually` should be mentioned as well for people that want to review each change. I would not be afraid to put more text into this dialog given its importance

**Editor**

![image](https://user-images.githubusercontent.com/900690/87802134-05986100-c851-11ea-8da6-ebde3acae416.png)

I would think having buttons inside the editor for performing an action would help, even if you can do the same from the list. E.g. you could have a blue `Accept` remote on the left and a `Accept Local` on the right. This would also make it easier to understand which side is which origin.

**Merge Flow**

![image](https://user-images.githubusercontent.com/900690/87802324-4f814700-c851-11ea-938c-b4e205ce3f60.png)

This flow is really confusing to me. I would have expected that when I click "Merge", the "magic" will happen without me having to do anything, why do I still have to "Accept Remote" or "Accept Merge Prview"? Isn't that redundant with the other actions?

**Icons**

The merge icon is still wrong I think. It should be the other way around (2 things getting merged into 1).

**Comment by sandy081** [comment]
Thanks @bpasero for the feedback

**Dialog**

<img width="630" alt="Screenshot 2020-07-18 at 09 49 46" src="https://user-images.githubusercontent.com/10746682/87848656-87849a80-c8e2-11ea-8c96-8475e826417a.png">

Changed the dialog and wordings. User input is not needed for merge action

**Editor**

<img width="816" alt="Screenshot 2020-07-18 at 10 28 54" src="https://user-images.githubusercontent.com/10746682/87848678-b995fc80-c8e2-11ea-91cb-09574edc8a11.png">

@misolori also mentioned the same. Added buttons for more visibility

**Merge Flow**

It seems you hit conflicts while merging, hence diff editor is shown to resolve conflicts and accept merges. Otherwise merge will happen without any user action. I see it is not clear that user hit conflicts on merge, so I added warning notification and updated the diff editor buttons

<img width="581" alt="Screenshot 2020-07-18 at 10 42 24" src="https://user-images.githubusercontent.com/10746682/87848777-712b0e80-c8e3-11ea-95e4-4c1d51cd905b.png">

<img width="765" alt="Screenshot 2020-07-18 at 10 42 42" src="https://user-images.githubusercontent.com/10746682/87848785-75efc280-c8e3-11ea-8695-f2e65d9c8c07.png">

Hope this helps.

**Icons**

@misolori  How about following icon for merge?

<img width="431" alt="image" src="https://user-images.githubusercontent.com/10746682/87848802-9d468f80-c8e3-11ea-8e1b-7ed099c09f2d.png">

**Comment by bpasero** [comment]
@sandy081 some more thoughts:
* when I clicked "Sync Manually..." I think we need to explain to the user what the user is supposed to do. Would it make sense to bring up another modal dialog with just a "OK" button that gives some information such as "you need to go through each change and click one of the three buttons"? Could even have a link to some online docs once we have it
* after I click one of the three buttons, the list item is no longer selected, I suggest to reselect so that the user knows what is opened in the editor area
* I wonder if there should be a way to see what choice I made (Local, Remote, Merge) at the list item level, currently there is just a checkmark but no way for me to know what I picked
* the merge flow is still tough on me when I have a conflict, that little notification is too subtle imho

**Comment by sandy081** [comment]
> when I clicked "Sync Manually..." I think we need to explain to the user what the user is supposed to do. Would it make sense to bring up another modal dialog with just a "OK" button that gives some information such as "you need to go through each change and click one of the three buttons"? Could even have a link to some online docs once we have it

Not a big fan of back to back dialogs. But I agree a brief explanation about manual sync is helpful. How about having this message inside the view, say above changes? @misolori what do you think?

> the merge flow is still tough on me when I have a conflict, that little notification is too subtle imho
 
Have to think what can be done here. Any suggestions are welcome.

**Comment by bpasero** [comment]
> How about having this message inside the view, say above changes

Yes, that is a good solution 👍 

> Edit: How about changing the notification to dialog?

Yes, much better. Since this is a result of a user action, going with dialog makes  a lot of sense to me.

**Comment by sandy081** [comment]
View shows following message.

<img width="1098" alt="image" src="https://user-images.githubusercontent.com/10746682/87914709-57b7cd00-ca71-11ea-8bb3-69b11b99eae1.png">

When there are conflicts while merging above dialog is shown. Note that diff editor is also opened in the background.

**Comment by sandy081** [comment]
> I wonder if there should be a way to see what choice I made (Local, Remote, Merge) at the list item level, currently there is just a checkmark but no way for me to know what I picked

I do not know if this is really needed as I guess user is aware of the action. I would wait for more user input before going with this.

**Comment by misolori** [comment]
Here's a couple of variations for the merge icon:

![image](https://user-images.githubusercontent.com/35271042/87992037-35877480-ca9c-11ea-907b-f5db787d5609.png)

**Comment by sandy081** [comment]
I like option 4.

**Comment by bpasero** [comment]
Do we not typically show the welcome text below the list and above the buttons? I feel it looks a bit weird to have the list appear below some text and not directly below the header.

**Comment by misolori** [comment]
@sandy081 this is now added via https://github.com/microsoft/vscode/commit/f0646f81aa1c4ba5cdada749ca1a18bf60b14dc1 and you can use `merge`.

Personally, the text feels really out of place here. Since there's only view here, why isn't this merged with the title?

![image](https://user-images.githubusercontent.com/35271042/88017659-f6c4df00-cada-11ea-82dd-887869e4cc0a.png)

**Comment by sandy081** [comment]
- updated the merge icon
- merged the view title with view container

<img width="358" alt="image" src="https://user-images.githubusercontent.com/10746682/88020592-513a5b80-cb2c-11ea-95be-9197fa22d388.png">

We already show summary text in the top in custom views, eg: references view. 

This is how it looks when text is moved below the list and above the buttons

<img width="369" alt="image" src="https://user-images.githubusercontent.com/10746682/88021074-1edd2e00-cb2d-11ea-8e3a-d96ff89c20c4.png">

**Comment by bpasero** [comment]
@sandy081 

**Dialog**

I would remove the trailing dot here:

![image](https://user-images.githubusercontent.com/900690/88062062-9b413280-cb68-11ea-9ebe-23e9373aa890.png)

**List**

This looks a bit unbalanced (the checkmark) to the rest of the actions:

![image](https://user-images.githubusercontent.com/900690/88062183-c62b8680-cb68-11ea-81ef-108f427c0acd.png)

I think I got no more feedback, this seems solid to me. We can always polish more, but I think the flow is understandable. I would put it on the test plan though to gather other peoples opinions.

**Comment by misolori** [comment]
Regarding the list, @sandy081 is the check a codicon or Unicode character? It looks really small like it’s using the decorator styles. Maybe we can make that normal size + a codicon?

**Comment by sandy081** [comment]
Thanks a lot for feedback @bpasero .

I renamed the view to **Merges** and updated the wordings and actions in the dialog accordingly. Calling this `Manual Merge...` and `Merges` aligns very well with the flow - While turning on sync we detect that there are changes from another machine and we offer user to `Merge` (auto) , or `Replace Local`  or `Merge Manually..`. User can choose to Merge manually and merge all changes and enable sync.

<img width="627" alt="Screenshot 2020-07-22 at 09 26 45" src="https://user-images.githubusercontent.com/10746682/88148025-4ea83680-cbfe-11ea-80c7-495842b2c601.png">

<img width="385" alt="Screenshot 2020-07-22 at 09 28 35" src="https://user-images.githubusercontent.com/10746682/88148009-494aec00-cbfe-11ea-911e-29f642bb6746.png">

**Comment by sandy081** [comment]
@misolori Yes, checkmark is a letter (unicode character) decoration and I think decorations do not support codicons.

**Comment by sandy081** [comment]
Closing this and please open separate issues for other feedback.
